core/src/main/java/org/apache/gravitino/listener/api/event/GrantUserRolesPreEvent.java

```diff
@@ -41,7 +44,7 @@ protected GrantUserRolesPreEvent(
     super(initiator, NameIdentifierUtil.ofUser(metalake, userName));
 
     this.userName = userName;
-    this.roles = roles;
+    this.roles = ImmutableList.copyOf(roles);
```

Comment:
no need to make roles immutable in pre event, as one of the purpose of the pre event is to allow user to change the request.